src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -25,10 +25,16 @@ void JitReadTuples::before_query(const Table& in_table, JitRuntimeContext& conte
   // Copy all input literals to the runtime tuple
   for (const auto& input_literal : _input_literals) {
     auto data_type = input_literal.tuple_value.data_type();
-    resolve_data_type(data_type, [&](auto type) {
-      using DataType = typename decltype(type)::type;
-      context.tuple.set<DataType>(input_literal.tuple_value.tuple_index(), boost::get<DataType>(input_literal.value));
-    });
+    if (data_type != DataType::Null) {
+      resolve_data_type(data_type, [&](auto type) {
+        using DataType = typename decltype(type)::type;
+        context.tuple.set<DataType>(input_literal.tuple_value.tuple_index(), boost::get<DataType>(input_literal.value));
+        // Non-jit operators store bool values as int values
+        if constexpr (std::is_same_v<DataType, int32_t>) {
```

Comment:
I proposed to @mrzzzrm to add a named type for these 'bool-hacks', but I can't find it anywhere. Did that happen?

Reply:
https://github.com/hyrise/hyrise/blob/master/src/lib/expression/evaluation/expression_evaluator.hpp

If we need that alias globally it should probably go into types.hpp

Reply:
...or all_type_variants.hpp, where all other data types are. Not that I particularly like that place, but it feels more consistent.

Reply:
fine with me